day-one-challenge/tyranny-of-the-rocket/getFuelRequirements.js

```diff
@@ -0,0 +1,41 @@
+const fs = require('fs');
+const path = require('path')
+const text = path.join(__dirname, 'module-mass.txt')
+const massArray = fs.readFileSync(path.join(text)).toString('utf-8').split("\n");
+
+//individually calculate the fuel needed for each mass of each module
+//to calculate the individual fuel requirements divide the mass by 3, then round down, and subtract 2
+//then sum together all the fuel requirements 
+//each fuel also requires it's own fuel
+
+const findfuel = num => {
+  return Math.floor(num / 3) - 2;
+};
+
+const totalfuel = array => {
+  let sum = 0
+
+  let fuelArray = [];
+  // console.log(array);
+  // console.log(array.length);
+  for (let index = 0; index < array.length; index++) {
+    let currentMass = (parseInt(array[index]));
+    let totalfuel = 0;
+
+    while (findfuel(currentMass) > 0) {
+      totalfuel += findfuel(currentMass);
+      currentMass = findfuel(currentMass);
+    }
+    fuelArray.push(totalfuel);
+  }
+
+  for (let index = 0; index < fuelArray.length; index++) {
+    sum += fuelArray[index];
+  };
+
+  return sum;
+}
```

Comment:
I think this is a very nice imperative way of solving the fuel calculation, @susmitabhowmik! I do remember that you said that you were going to go back and experiment with a more functional solution.

That being said, if this were to be the final implementation then I would consider finding a way not to call `findfuel()` so many times in lines 25-27. Also, at line 22 I think if you were to have the data already converted during the I/O phase suggested above you wouldn't have to worry about converting it here inside your actual calculation.